src/Estimators/include/BipedalLocomotion/Estimators/FloatingBaseEstimator.h

```diff
@@ -0,0 +1,280 @@
+    * @param[in] model reduced iDynTree model required by the estimator
+    * @note any custom initialization of parameters or the algorithm implementation is not done here,
+    *       it must be done in customInitialization() by the child class implementing the algorithm
+    * @return True in case of success, false otherwise.
+    */
+    bool initialize(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler, const iDynTree::Model& model);
+
+    /**
+    * Set the polled IMU measurement
+    * @param[in] acc_meas linear accelerometer measurement expressed in the local IMU frame (m/s^2)
+    * @param[in] gyro_meas gyroscope measurement expressed in the local IMU frame (rad/s)
+    * @return True in case of success, false otherwise.
+    */
+    bool setIMUMeasurement(const Eigen::Vector3d& acc_meas,
+                           const Eigen::Vector3d& gyro_meas);
+
+    /**
+    * Set feet contact states
+    * @param[in] lf_in_contact left foot contact state [0, 1]
+    * @param[in] rf_in_contact right foot contact state [0, 1]
+    * @return True in case of success, false otherwise.
+    */
+    bool setContacts(const bool& lf_in_contact, const bool& rf_in_contact);
+
+    /**
+    * Set kinematic measurements
+    * @note it is assumed that the order of the joints loaded in the model and the order of the measurements in these vectors match
+    * @param[in] encoders joint positions measured through encoders
+    * @param[in] encoder_speeds joint velocities measured through encoders
+    * @return True in case of success, false otherwise.
+    */
+    bool setKinematics(const Eigen::VectorXd& encoders,
+                       const Eigen::VectorXd& encoder_speeds);
+
+    /**
+    * Compute one step of the estimator
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool advance() final;
+
+    /**
+    * Get estimator outputs
+    * @return A struct containing he estimated internal states of the estiamtor and the associated covariance matrix
+    */
+    virtual const FloatingBaseEstimators::Output& get() const final;
+
+    /**
+    * Determines the validity of the object retrieved with get()
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool isValid() const final { return (m_estimator_state == State::Running); };
+
+    /**
+    * Get ModelComputations object by reference
+    * @return ModelComputations object providing information between considered model related quantities in the estimator
+    * like the base link, IMU, feet contact frames.
+    */
+    ModelComputations& modelComputations();
+
+protected:
+    /**
+    * Configure generic parameters
+    * @param[in] handler configure the generic parameters for the estimator
+    * @return True in case of success, false otherwise.
+    */
+    bool initialize(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler);
```

Comment:
Notice that if you inherit this from a base class (even if it is still not the case), moving from ``public`` to ``private`` is not ideal. If you have a pointer to the base class, you can still access this method because it is public for the base class. Hence, in my opinion, either you move it to the public section as before, or you remove it.

Reply:
The intention was to keep it private for the moment and make it public when inheriting from the `Block` class, and indicating the `modelComputations().setModel()` method as a prior step, if ever this function was called.

If you agree, I will keep it as is for the moment, do the necessary changes in the future. Otherwise, I can make the change already. Let me know what you think.

Reply:
Ok, make sense, no need to change it now.